source/common/common/utility.cc

```diff
@@ -221,6 +223,15 @@ std::string DateFormatter::now(TimeSource& time_source) {
   return fromTime(time_source.systemTime());
 }
 
+FixedSizeStreamBuffer::FixedSizeStreamBuffer(char* base, size_t size) {
+  this->setp(base, base + size);
+}
+
+OutputBufferStream::OutputBufferStream(char* data, size_t size)
+    : FixedSizeStreamBuffer{data, size}, std::ostream{static_cast<std::streambuf*>(this)} {}
```

Comment:
Does the call to setp need to happen before a pointer is given to the ostream constructor?

This seems to work:
`/**`
` * std::ostream class that serializes writes into the provided buffer.`
` */`
`class OutputBufferStream : private std::streambuf, public std::ostream {`
`public:`
`  OutputBufferStream(char* data, size_t size);`
`...`
`}`

`OutputBufferStream::OutputBufferStream(char* data, size_t size)`
`    : std::ostream(this) {`
`  setp(data, data + size);`
`}`

Reply:
The call to setp doesn't need to occur first since streambuf are more focused on managing some storage (buffer, file, ...) and ostreams are more focused on  serializing objects. ostream defers the actual writing / storage management to the streambuf.

I organized it like this to be consistent with the `InputConstMemoryStream`. If you want I can flatten it (removing `FixedSizeStreamBuffer`)

Reply:
I would consider using the name MutableMemoryStreamBuffer

I don't feel strongly about flattening it.  Flattening it seems slightly cleaner but both approaches are fine.

Reply:
Done.